src/coreclr/vm/corhost.cpp

```diff
@@ -359,6 +365,11 @@ HRESULT CorHost2::ExecuteAssembly(DWORD dwAppDomainId,
     UNINSTALL_UNWIND_AND_CONTINUE_HANDLER;
     UNINSTALL_UNHANDLED_MANAGED_EXCEPTION_TRAP;
 
+    {
+        GCX_COOP();
+        unhandledExceptionMarkerFrame.Pop(pThread);
+    }
+
 #ifdef LOG_EXECUTABLE_ALLOCATOR_STATISTICS
     ExecutableAllocator::DumpHolderUsage();
     ExecutionManager::DumpExecutionManagerUsage();
```

Comment:
Does ExecuteInDefaultAppDomain below need the same change?

Reply:
Yes, I've missed that one.